src/scheduler.c

```diff
@@ -74,20 +78,41 @@ void *scheduler_thread(void *args)
   * either file_changed or a condition in the schedule 
   */
         struct timespec tm;
+        int info_period = 3;
     
         if (0 == clock_gettime(CLOCK_REALTIME, &tm) && current_schedule) {
+            static char *current_blocked_macs = NULL;
             char *blocked_macs;
-            time_t unix_time = tm.tv_sec; // ignore?  + (tm.tv_nsec / 1000000000)
+            time_t unix_time = tm.tv_sec; // ignore?  +(tm.tv_nsec / 1000000000)
             blocked_macs = get_blocked_at_time(current_schedule, unix_time);
+             
+            if (NULL == current_blocked_macs) {
+                    current_blocked_macs = copy_new_mac_address_list(
```

Comment:
You don't want to do this.  `get_blocked_at_time()` returns you a malloced buffer to own and care for that you can simply directly use.